modules/Microsoft.Compute/galleries/application/deploy.bicep

```
@@ -0,0 +1,100 @@
+@description('Required. Name of the application definition.')
+param name string
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Conditional. The name of the parent Azure Compute Gallery. Required if the template is used in a standalone deployment.')
+@minLength(1)
+param galleryName string
+
+@description('Optional. The description of this gallery Application Definition resource. This property is updatable.')
+param applicationDefinitionDescription string = ''
+
+@description('Optional. The Eula agreement for the gallery Application Definition. Has to be a valid URL.')
+param eula string = ''
+
+@description('Optional. The privacy statement uri. Has to be a valid URL.')
+param privacyStatementUri string = ''
+
+@description('Optional. The release note uri. Has to be a valid URL.')
+param releaseNoteUri string = ''
+
+@description('Optional. This property allows you to specify the supported type of the OS that application is built for.')
+@allowed([
+  'Windows'
+  'Linux'
+])
+param supportedOSType string = 'Windows'
+
+@description('Optional. The end of life date of the gallery Image Definition. This property can be used for decommissioning purposes. This property is updatable. Allowed format: 2020-01-10T23:00:00.000Z.')
+param endOfLife string = ''
```

I'd suggest to keep the original property name 
```suggestion
param endOfLifeDate string = ''
```

---

fixed